src/yaml-language-support/yaml-support.ts

```diff
@@ -0,0 +1,68 @@
+import * as vscode from 'vscode';
+
+import { parse, findNodeAtPosition, util as yamlUtil } from 'node-yaml-parser';
+
+export interface YamlCache {
+    // the document model list represents the yaml text
+    yamlDocs: any[];
+
+    // lineLens contains the converted line length of each lines, it is used for converting from
+    // vscode position to the inner position in yaml element model.
+    lineLens: number[];
```

Comment:
I read this as 'lens' (as in CodeLens).  `lineLengths` would be clearer.

Reply:
sure, I will change it to lineLengths later